src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static int DEFAULT_SIZE;
+    private static int RESIZE_FACTOR;
+    private int size;
+    private T[] elements;
+
+    public ArrayList() {
+        DEFAULT_SIZE = 10;
+        RESIZE_FACTOR = 5;
```

Comment:
only non-static fields of the class should be initialized in a constructor, make these final, init at declaration
resize factor 5 implies you are growing it x5 times